Entity/MapRepository.php

```diff
@@ -127,4 +137,62 @@ public function findOneForUrlOrPath($url, $path)
         return current($maps);
     }
 
-}
+    /**
+     * Get groups.
+     *
+     * @return array
+     */
+    private function getGroups()
+    {
+        $groups =  $this->_em->getConnection()->fetchAll('SELECT id, priority FROM armb_redirect_group');
+        return  \array_reduce($groups, function ($result, $item) {
+            $result[$item['id']] = $item['priority'];
+            return $result;
+        }, []);
+
+    }
+
+    /**
+     * Build map from array.
+     *
+     * @param array $data
+     * @return Map
+     * @throws \Exception
+     */
+    public function buildMapFromArray(array $data)
```

Comment:
@micromax14  This should not be in a Repository class?

Reply:
This should be an toArray method on the entity class

Reply:
yes, I forgot to move that. thaks